rosa/cli.py

```diff
@@ -160,98 +160,40 @@ def get_available_flags(command):
-    _commands = get_available_commands(command=[rosa_cmd])
-    output_flag_str = "-o, --output"
-    auto_answer_yes_str = "-y, --yes"
-    auto_mode_str = "-m, --mode"
-    region_str = "--region"
-
-    for command in _commands:
-        commands_dict.setdefault(command, {})
-
-    for top_command in commands_dict.keys():
-        # Always get top commands options
-        commands_dict[top_command]["json_output"] = check_flag_in_flags(
-            command_list=[rosa_cmd, top_command],
-            flag_str=output_flag_str,
-        )
-        commands_dict[top_command]["auto_answer_yes"] = check_flag_in_flags(
-            command_list=[rosa_cmd, top_command],
-            flag_str=auto_answer_yes_str,
-        )
-        commands_dict[top_command]["auto_mode"] = check_flag_in_flags(
-            command_list=[rosa_cmd, top_command],
-            flag_str=auto_mode_str,
-        )
-        commands_dict[top_command]["region"] = check_flag_in_flags(
-            command_list=[rosa_cmd, top_command],
-            flag_str=region_str,
-        )
+def parse_help():
+    def _fill_commands_dict_with_support_flags(commands_dict, flag_key_path):
+        support_commands = {
+            "json_output": "-o, --output",
+            "auto_answer_yes": "-y, --yes",
+            "auto_mode": "-m, --mode",
+            "region": "--region",
+        }
+        for cli_flag, flag_value in support_commands.items():
+            commands_dict[flag_key_path][cli_flag] = check_flag_in_flags(
+                command_list=["rosa"] + flag_key_path,
+                flag_str=flag_value,
+            )
+        return commands_dict
 
-        _commands = get_available_commands(command=[rosa_cmd, top_command])
-
-        if _commands:
-            for command in _commands:
-                commands_dict[top_command][command] = {}
-                # Always get sub commands options
-                commands_dict[top_command][command][
-                    "json_output"
-                ] = check_flag_in_flags(
-                    command_list=[rosa_cmd, top_command, command],
-                    flag_str=output_flag_str,
-                )
-                commands_dict[top_command][command][
-                    "auto_answer_yes"
-                ] = check_flag_in_flags(
-                    command_list=[rosa_cmd, top_command, command],
-                    flag_str=auto_answer_yes_str,
-                )
-                commands_dict[top_command][command]["auto_mode"] = check_flag_in_flags(
-                    command_list=[rosa_cmd, top_command, command],
-                    flag_str=auto_mode_str,
-                )
-                commands_dict[top_command][command]["region"] = check_flag_in_flags(
-                    command_list=[rosa_cmd, top_command, command],
-                    flag_str=region_str,
-                )
+    def _build_command_tree(commands_dict, commands_search_path=None):
+        _commands_dict = commands_dict
```

Comment:
`_commands_dict = commands_dict` why you assign it?
It will be a reference to the same dict

Reply:
Just because I thought about avoiding calling with `commands_dict=commands_dict` in line [190](https://github.com/RedHatQE/rosa-python-client/pull/34/files#diff-d727c959dd93ddbb1b1b2d0cd8124ec5d587ebc047afe9f117e2d7e7858fbd23R190)
What do you think?

Reply:
This assignment is useless and unnecessary, it's just point to the same object.

Reply:
Done